### PR TITLE
Deprecation check for No Master Block setting

### PR DESCRIPTION
**Pull request by AthenaEryma**
`discovery.zen.no_master_block` is deprecated in favor of
`cluster.no_master_block` in 7.0, but is a dynamic setting so we need
both node- and cluster-level checks for it.

Relates to https://github.com/elastic/elasticsearch/pull/38350 and https://github.com/elastic/elasticsearch/issues/36024

### CODE REVIEW

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java [hunk 1]
```diff
@@ -32,6 +33,17 @@ static DeprecationIssue httpEnabledSettingRemoved(Settings nodeSettings, Plugins
         return null;
     }
 
+    static DeprecationIssue noMasterBlockRenamed(Settings nodeSettings, PluginsAndModules plugins) {
+        if (nodeSettings.hasValue(NO_MASTER_BLOCK_SETTING.getKey())) {
+            return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
+                "Master block setting renamed",
+                "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-7.0.html" +
+                    "_new_name_for_literal_no_maaster_block_literal_setting",
```
Comment (DaveCTurner) [inline x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: Double `a` in `maaster`.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I would interpret the message here as indicating that I can update my settings in 6.x, but that's not the case. I think we should be clearer that we will quietly _ignore_ `discovery.zen.no_master_block` in 7.0, and the action that the user should take is to clear any value set for `discovery.zen.no_master_block` pre-upgrade and to set `cluster.no_master_block` post-upgrade.

**Comment by gwbrown** [comment]
Thanks for the review @DaveCTurner, I've addressed your comments - what do you think of the new message?

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**